sbt/src/sbt-test/compiler-project/inc-pickled-existential/build.sbt

```diff
@@ -1,15 +1,28 @@
 import sbt.internal.inc.Analysis
+import complete.DefaultParsers._
 
 logLevel := Level.Debug
 
-// dumps analysis into target/analysis-dump.txt file
-InputKey[Unit]("check-number-of-compiler-iterations") := {
-  val args = Def.spaceDelimited().parsed
-  val a = (compile in Compile).value.asInstanceOf[Analysis]
-  assert(args.size == 1)
-  val expectedIterationsNumber = args(0).toInt
-  assert(
-    a.compilations.allCompilations.size == expectedIterationsNumber,
-    "a.compilations.allCompilations.size = %d (expected %d)".format(
-      a.compilations.allCompilations.size, expectedIterationsNumber))
+// Reset compiler iterations, necessary because tests run in batch mode
+val recordPreviousIterations = taskKey[Unit]("Record previous iterations.")
+recordPreviousIterations := {
+  CompileState.previousIterations = {
+    val previousAnalysis = (previousCompile in Compile).value.analysis
+    if (previousAnalysis.isEmpty) {
+      streams.value.log.info("No previous analysis detected")
+      0
+    } else {
+      previousAnalysis.get match {
+        case a: Analysis => a.compilations.allCompilations.size
+      }
+    }
+  }
+}
+
+val checkIterations = inputKey[Unit]("Verifies the accumulated number of iterations of incremental compilation.")
+
+checkIterations := {
+  val expected: Int = (Space ~> NatBasic).parsed
+  val actual: Int = ((compile in Compile).value match { case a: Analysis => a.compilations.allCompilations.size }) - CompileState.previousIterations
+  assert(expected == actual, s"Expected $expected compilations, got $actual")
```

Comment:
This one should be moved to, shouldn't it? I think zinc's scripted supports this kind of assertion.

There are certainly other tests that could be moved over to zinc. Is there a reason to keep them here?